cas/service.c

```diff
@@ -1011,8 +1015,36 @@ static void cas_fom_failure(struct cas_fom *fom, int rc, bool ctg_op_fini)
 	m0_fom_phase_move(&fom->cf_fom, rc, M0_FOPH_FAILURE);
 }
 
+static int cas_dtm0_logrec_add (struct m0_fom *fom0, struct m0_dtm0_tx_desc *txd,
+				enum m0_dtm0_tx_pa_state state) {
+	/* log the dtm0 logrec before completing the cas op */
+	struct m0_dtm0_service *dtms = m0_dtm0_service_find(fom0->fo_service->rs_reqh);
+	struct m0_dtm0_tx_desc *msg = &cas_op(fom0)->cg_txd;
+	struct m0_buf	        buf = {};
+	int                     i;
+	int			rc;
+
+	for (i = 0; i < msg->dtd_ps.dtp_nr; ++i) {
+		if (m0_fid_eq(&msg->dtd_ps.dtp_pa[i].p_fid,
+					&dtms->dos_generic.rs_service_fid)) {
+			msg->dtd_ps.dtp_pa[i].p_state = (uint32_t) M0_DTPS_PERSISTENT;
+			break;
+		}
+	}
+	rc = m0_xcode_obj_enc_to_buf(&M0_XCODE_OBJ(m0_cas_op_xc, cas_op(fom0)),
+				     &buf.b_addr, &buf.b_nob);
+	M0_ASSERT(rc == 0);
```

Comment:
Can we handle the error here, instead of panic?

Reply:
Handled.